src/dtls13.c

```diff
@@ -956,8 +956,18 @@ static int Dtls13SendOneFragmentRtx(WOLFSSL* ssl,
         handshakeType, hashOutput, Dtls13SendNow(ssl, handshakeType));
 
     if (rtxRecord != NULL) {
-        if (ret == 0 || ret == WC_NO_ERR_TRACE(WANT_WRITE))
+        if (ret == 0 || ret == WC_NO_ERR_TRACE(WANT_WRITE)) {
+        #ifdef WOLFSSL_RW_THREADED
+            int lockRet = wc_LockMutex(&ssl->dtls13Rtx.mutex);
+            if (lockRet < 0) {
+                return lockRet;
+            }
+        #endif
             Dtls13RtxAddRecord(&ssl->dtls13Rtx, rtxRecord);
+        #ifdef WOLFSSL_RW_THREADED
+            wc_UnLockMutex(&ssl->dtls13Rtx.mutex);
+        #endif
+        }
```

Comment:
I still think its better to use a high level lock. That will avoid any other possible races as well as handle other post-hs messages.